src/main.cc

```diff
@@ -670,29 +676,43 @@ main(int argc, char **argv)
 			return 0;
 		case 'i':
 			/* Force interactive mode */
-			interactive = true;
+			opt_mask |= O_INTERACTIVE;
+			break;
+		case 'b':
```

Comment:
There is one case left unhandled - when there was `-b` option without modifiers, and output file provided. Patched tarantool rejects this case, while original luajit does handle it correctly:

```
✘-1 ~/tarantool/build [tsafin/luajit-hack-0 L|…17⚑ 8]
11:37 $ ./src/tarantool -b ../debug-getsources-0.lua debug-getsources.out
Save LuaJIT bytecode: luajit -b[options] input output
  -l        Only list bytecode.
  -s        Strip debug info (default).
  -g        Keep debug info.
  -n name   Set module name (default: auto-detect from input name).
  -t type   Set output file type (default: auto-detect from output name).
  -a arch   Override architecture for object files (default: native).
  -o os     Override OS for object files (default: native).
  -e chunk  Use chunk string as input.
  --        Stop handling options.
  -         Use stdin as input and/or stdout as output.

File types: c h obj o raw (default)
✔ ~/tarantool/build [tsafin/luajit-hack-0 L|…17⚑ 8]
11:38 $ luajit -b ../debug-getsources-0.lua debug-getsources.out
```

Reply:
Also, we better modify the reported message to include tarantool name instead of luajit.

Reply:
Also I'd like to highlight, that I want this patch landed to master before #7929 where I modify all the same `init.c` for introduction of `-d` option, and given `-b` and `-j` changes from this pull request the possible resultant code with `-d` added similarly would look more generic (and shorter).